### PR TITLE
V-menu fix segfault and make it faster by not copying every item.

### PR DESCRIPTION
**Pull request by OzoneH3**

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -8842,7 +8842,7 @@ point game::look_around(WINDOW *w_info, const point pairCoordsFirst)
 }
 
 bool lcmatch(const std::string &str, const std::string &findstr); // ui.cpp
-bool game::list_items_match(item &item, std::string sPattern)
+bool game::list_items_match(item *item, std::string sPattern)
```
Comment (BevapDin) [inline src/game.cpp]: Why that change, why does the reference not work?

Comment (OzoneH3) [inline src/game.cpp]: Hmm, don't know why I changed that, went a little over board without thinking I guess.

**Comment by KA101** [comment]
OK, I'll change it back in the merge then.

File: src/game.cpp [hunk 2]
```diff
@@ -9285,7 +9285,7 @@ int game::list_items(const int iLastState)
     bool addcategory = false;
     int iPage = 0;
     int iCatSortNum = 0;
-    item activeItem;
+    map_item_stack *activeItem;
```
Comment (KA101) [inline src/game.cpp]: I'm thinking this broke the build: Jenkins/clang is throwing a maybe_uninitialized error.  Will see about a bandaid but might want to look into a better fix.

**Comment by OzoneH3** [comment]
Weird, there were no errors for me with g++. Also the fix with the = NULL initialisation is ok, it can't get accessed before it is set the first time anyway.

**Pull request merged**